sdk/core/azure-core/src/main/java/com/azure/core/util/polling/LocationPollingStrategy.java

```diff
@@ -0,0 +1,179 @@
+import com.azure.core.http.rest.Response;
+import com.azure.core.implementation.TypeUtil;
+import com.azure.core.util.BinaryData;
+import com.azure.core.util.Context;
+import com.azure.core.util.logging.ClientLogger;
+import com.azure.core.util.serializer.JacksonAdapter;
+import com.azure.core.util.serializer.SerializerEncoding;
+import com.azure.core.util.serializer.TypeReference;
+import reactor.core.publisher.Mono;
+
+import java.time.Duration;
+
+/**
+ * Implements a Location polling strategy.
+ *
+ * @param <T> the {@link TypeReference} of the response type from a polling call, or BinaryData if raw response body
+ *            should be kept
+ * @param <U> the {@link TypeReference} of the final result object to deserialize into, or BinaryData if raw response
+ *            body should be kept
+ */
+public class LocationPollingStrategy<T, U> implements PollingStrategy<T, U> {
+    private static final String LOCATION = "Location";
+    private static final String REQUEST_URL = "requestURL";
+    private static final String HTTP_METHOD = "httpMethod";
+    private static final String RETRY_AFTER = "Retry-After";
+    private static final String POLL_RESPONSE_BODY = "pollResponseBody";
+
+    private final JacksonAdapter serializer = new JacksonAdapter();
+    private final ClientLogger logger = new ClientLogger(LocationPollingStrategy.class);
+
+    private final HttpPipeline httpPipeline;
+    private final Context context;
+
+    /**
+     * Creates an instance of the location polling strategy.
+     *
+     * @param httpPipeline an instance of {@link HttpPipeline} to send requests with
+     * @param context additional metadata to pass along with the request
+     */
+    public LocationPollingStrategy(
+            HttpPipeline httpPipeline,
+            Context context) {
+        this.httpPipeline = httpPipeline;
+        this.context = context;
+    }
+
+    @Override
+    public Mono<Boolean> canPoll(Response<?> initialResponse) {
+        HttpHeader locationHeader = initialResponse.getHeaders().get(LOCATION);
+        return Mono.just(locationHeader != null);
+    }
+
+    @SuppressWarnings("unchecked")
+    @Override
+    public Mono<LongRunningOperationStatus> onInitialResponse(Response<?> response, PollingContext<T> pollingContext,
+                                                   TypeReference<T> pollResponseType) {
+        HttpHeader locationHeader = response.getHeaders().get(LOCATION);
+        if (locationHeader != null) {
+            pollingContext.setData(LOCATION, locationHeader.getValue());
+        }
+        pollingContext.setData(HTTP_METHOD, response.getRequest().getHttpMethod().name());
+        pollingContext.setData(REQUEST_URL, response.getRequest().getUrl().toString());
+
+        if (response.getStatusCode() == 200
+                || response.getStatusCode() == 201
+                || response.getStatusCode() == 202
+                || response.getStatusCode() == 204) {
```

Comment:
All all these code valid in a long-running operation?

Reply:
Individual services may have specific rules, but in general, these should be accepted. (200 for stepwise polling, 201 is "created" for PUT/PATCH, 202 is "accepted" for POST, 204 is a valid status for DELETE)